src/api/user/user.controller.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { UserController } from './user.controller';
+import { CreateUserDto } from './user.dto';
+import { User } from './user.entity';
+import { UserService } from './user.service';
+
+describe('UserController', () => {
+  let controller: UserController;
+  let service: UserService;
+
+  const USER_REPOSITORY_TOKEN = getRepositoryToken(User);
+
+  beforeEach(async () => {
+    const UserServiceProvider = {
+      provide: UserService,
+      useFactory: () => ({
+        findAll: jest.fn(),
+        getUser: jest.fn(),
+        createUser: jest.fn(() => []),
+      }),
+    };
+
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [UserController],
+      providers: [
+        UserService,
+        UserServiceProvider,
+        {
+          provide: USER_REPOSITORY_TOKEN,
```

Comment:
At the controller level, you don't need to mock out the Repository. The service is already mocked so this is never being used.